R/plotDots.R

```diff
@@ -101,6 +111,18 @@ plotDots <- function(object, features, group = NULL, block=NULL, exprs_values =
         group.names <- colnames(ave)
     }
 
+    if (center) {
+        ave <- ave - rowMeans(ave)
+    }
+    if (scale) {
+        ave <- ave / apply(ave, 1, function(row) {
+            sqrt(sum(row^2) / (length(row) - 1))
+        })
+    }
+    if (!is.null(max_ave)) {
+        ave <- pmin(max_ave, ave)
```

Comment:
We could make this more consistent with how `plotHeatmap()` does it, using a `zlim` argument to control the color scale.    (You could probably even spin this out into a separate utility.)